### ISSUE TITLE
ignition: systemd unit mask fail when target exists

### ISSUE THREAD

**Issue by j-vizcaino** [coreos/bugs#1292]
When using `mask: true` and target file exists, Ignition fails.
This is caused by `os.Symlink` not wanting to overwrite the target.

One way around this would be:

```
if the target file exists
  if target links to /dev/null
     return nil
  remove file (return if err)
return os.Symlink
```

**Comment by crawford** [comment]
@j-vizcaino how did you run into this issue? Are you running Ignition multiple times?

**Comment by j-vizcaino** [comment]
You are right, the bug occurred during testing after multiple reboots, with
a masked systemd unit. The thing is, I thought Ignition should be
idempotent so this should not be a problem.

**Comment by crawford** [comment]
Ignition should be able to handle this (PR is up). I was just curious how you ran into it.
